modules/data.remote/R/call_MODIS.R

```diff
@@ -111,11 +166,39 @@ call_MODIS <- function(outfolder = ".", start_date, end_date, lat, lon, size = 0
     output$lat <- round(output$lat, 4)
     output$lon <- round(output$lon, 4)
     
-    fname <- paste(product, "_", band, "output_", start_date, "_", end_date, "_", lat, "_", lon, ".csv", sep = "")
-    fname <- paste0(outfolder, "/", fname)
-    write.csv(output, fname)
-    return(output)}
-
+    if (QC_filter == T)
+    {
+      output$qc == as.character(output$qc)
+      for (i in 1:nrow(output))
+      {
+        convert = paste(binaryLogic::as.binary(as.integer(output$qc[i]), n = 8), collapse = "")
+        output$qc[i] = substr(convert, nchar(convert)-2, nchar(convert))
+      }
+      good = which(output$qc == "000" | output$qc == "001")
+      if (length(good) > 0 | !(is.null(good)))
+      {
+        output = output[good,]
+      } else {
+        print("All QC values are bad. No data to output with QC filter == TRUE.")
+      }
+    }
+    
+    if (!(outfolder == "") & !(is.null(siteID)))
+    {
+      fname <- paste(product, "_", band, "_", siteID, "_output_", start_date, "_", end_date, "_", spatial.tools::add_leading_zeroes(iter, 4), ".csv", sep = "")
+      fname <- paste0(outfolder, "/", fname)
+      write.csv(output, fname, row.names = F)
+    }
+    if (!(outfolder == "") & is.null(siteID))
+    {
+      fname <- paste(product, "_", band, "_output_", lat, "_", lon, "_", start_date, "_", end_date, "_", spatial.tools::add_leading_zeroes(iter, 2), ".csv", sep = "")
```

Comment:
Again, replace the `add_leading_zeroes` with `sprintf("%02d")`.

Reply:
Just out of curiosity, is sprintf("%02d") like a regex type function that does the exact same thing as add_leading_zeroes?

Reply:
never used it before so it's handy if so

Reply:
Not quite. AFAIK, it's based on C's string formatting functions. My syntax above was wrong -- you really want `sprintf("%02d", iter)`.

Here are some more examples:

``` r
sprintf("An AU is %d miles", 93e6)
#> [1] "An AU is 93000000 miles"
sprintf("The current date is %d-%02d-%02d", 2019, 5, 9)
#> [1] "The current date is 2019-05-09"
sprintf("The price of %s is $%.2f per pound.",
        c("apples", "oranges", "bananas"),
        c(1.78912312, 1.6912341, 0.12341))
#> [1] "The price of apples is $1.79 per pound." 
#> [2] "The price of oranges is $1.69 per pound."
#> [3] "The price of bananas is $0.12 per pound."
```

There's a lot more you can do with formatting codes -- see [`?sprintf`](https://www.rdocumentation.org/packages/base/versions/3.6.0/topics/sprintf) for more details.

Reply:
The most relevant one for you from that group is the second one ("current date"). `%02d` is parsed out as:

- `%` -- this indicates the beginning of  a replacement format code
- `02` -- The `0` means "pad with zeros", and `2` indicates how many to pad with. To pad with 10 zeros, you would use `010`. Without the leading `0` (i.e. just `%2d`), pad with spaces instead, so `sprintf("%5d", 5)` would make `     5`.
- `d` -- This means the argument is an integer. Other common ones are `s` for "string" and `f` for float. Different inputs have different formatting types. The padding stuff works for `s`, `d`, and `f` (though a little differently for `f`), but for `f` you can also specify the precision after the decimal point, for example (as I do in the last example above).

Reply:
thanks! fixed also